src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Services/DataSetVersionMappingService.cs

```diff
@@ -63,6 +63,21 @@ public async Task<Either<ActionResult, Unit>> CreateMappings(
             sourceFilterMeta,
             nextVersionMeta.Filters);
 
+        var hasDeletedIndicators = await HasDeletedIndicators(
+            liveVersion.Id,
+            nextVersionMeta.Indicators,
+            cancellationToken);
+
+        var hasDeletedGeographicLevels = await HasDeletedGeographicLevels(
```

Comment:
I think we can infer this one currently from the fact that we have a LocationLevelMapping for a particular location level in the plan that has no Candidates.  I'm not against storing it explicitly for time-saving purposes, but just wondering it there was something I missed?

Reply:
In fact, you have it in the code for `IsMajorVersionUpdate()`:

```
        var hasDeletedLocationLevels = mapping.LocationMappingPlan
            .Levels
            .Any(level => level.Value.Candidates.Count == 0);
```

So I was wondering what the benefit of having it in a separate column is, unless it doesn't represent the exact same thing?

Reply:
Location levels and geographic levels aren't the same thing though:

- location levels correspond to CSV columns
- geographic levels correspond to the values specifically in the `geographic_level` column

Whilst similar looking, these aren't the same thing. For example, a data set may have location levels for national (country), regional and local authority, but may only contain data where the `geographic_level` column is `Local authority` (i.e. only local authority level data).

This means there are:

- location levels - NAT, REG, LA
- geographic levels - LA

This is a really subtle difference, but is entirely possible and needs to be encoded correctly. The location mapping plan only considers location levels (i.e. `LocationMeta` and `LocationOptionMeta`), whereas we have a separate `GeographicLevelMeta` that handles geographic levels.

Reply:
Gotcha, makes sense now, and at some point I'm going to get this concept...